// @ckeditor/ckeditor5-redact/src/redactediting.js
import { Plugin } from 'ckeditor5/src/core';
import AttributeCommand from '@ckeditor/ckeditor5-basic-styles/src/attributecommand';
const REDACT = 'redact';
export default class RedactEditing extends Plugin {
    init() {
        this._defineSchema();
        this._defineConverters();
    }
    _defineSchema() {
        console.log('RedactEditing#init() got called');
        const schema = this.editor.model.schema;
        // Extend the text node's schema to accept the redact attribute.
        schema.extend('$text', {
            allowAttributes: [REDACT]
        });
        schema.setAttributeProperties(REDACT, {
            isFormatting: true,
            copyOnEnter: true
        });
    }
    _defineConverters() {
        const editor = this.editor;
        // Conversion from a model attribute to a view element
        editor.conversion.attributeToElement({
            model: REDACT,
            view: {
                name: 'del',
                classes: 'redacted',
                styles: { 'color': 'red', 'background': 'papayawhip' }
            },
            upcastAlso: [
                {
                    styles: {
                        'text-decoration': 'line-through'
                    }
                }
            ]
        });
        // Create strikethrough command.
        editor.commands.add(REDACT, new AttributeCommand(editor, REDACT));
        // Set the Ctrl+j keystroke.
        // Note: Not all keys are usable such as period/greater than.
        editor.keystrokes.set('CTRL+j', REDACT);
    }
}
